### PR TITLE
Initial Commit (course view)

### PR DESCRIPTION
**Pull request by Mahmoud-Abdelaty**
## Changelog

I have successfully completed the task, which is the course view. I hope you like my work.
if u need any changes tell me bro 👍

### CODE REVIEW

**Review by moha-b — CHANGES_REQUESTED** [review]
### Great job on the UI! 👏 
I appreciate the attention to detail, and it's awesome to see such a pixel-perfect design. However, I have a few small suggestions to make it even better:

- [x] Double-check the unit of measurement used on some widgets `sp` was used instead of `w` or `h`.
- [ ] don't forget to return the main to the idle case

Feel free to reach out if you have any questions or need clarification on these tweaks. Your dedication to achieving pixel-perfection is truly commendable!

Keep up the fantastic work! 🌟

File: lib/course_view/data/models/course/course_status_model.dart
Comment (moha-b) [inline lib/course_view/data/models/course/course_status_model.dart:1]: well done 👍

**Review by moha-b — CHANGES_REQUESTED** [review]
Small adjustments like these contribute to a more polished and consistent UI. Give it a try, and let me know your thoughts or if you have any questions!

File: lib/course_view/view/widgets/video_course.dart [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    return Padding(
+      padding: EdgeInsets.fromLTRB(24.w, 24.h, 24.w, 0.h),
+      child: Stack(
+        children: [
+          Container(
+            width: 327.w,
+            height: 171.h,
+            foregroundDecoration: BoxDecoration(
+              color: AppColors.black.withOpacity(0.6),
+              borderRadius: BorderRadius.circular(15),
```
[the remark this answers is not in the document]
Comment (Mahmoud-Abdelaty) [reply lib/course_view/view/widgets/video_course.dart]: Done ??

**Comment by Mahmoud-Abdelaty** [comment]
Done ?

File: lib/course_view/view/widgets/course_data_section.dart [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                  'Power BI advanced Data Analysis',
+                  style: TextStyle(
+                    color: AppColors.gray900,
+                    fontSize: 24.sp,
+                    fontWeight: FontWeight.w700,
+                  ),
+                ),
+              ),
+              const Spacer(),
+              GestureDetector(
```
[the remark this answers is not in the document]
Comment (Mahmoud-Abdelaty) [reply lib/course_view/view/widgets/course_data_section.dart:44]: test

[the remark this answers is not in the document]
Comment (Mahmoud-Abdelaty) [reply lib/course_view/view/widgets/course_data_section.dart:44]: > Have you considered using an `IconButton` instead? It brings a more native feel to the interaction.

ee

**Review by moha-b — COMMENTED** [review]
sry I forgot that

File: lib/course_view/view/widgets/video_course.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (moha-b) [reply lib/course_view/view/widgets/video_course.dart]: it's okay mahmoud

**Pull request merged**